In [157]:
import pandas as pd 
import numpy as np

In [158]:
YEAR_CSV_PATH = '../data/YEAR.csv'
MONTH_CSV_PATH = '../data/MONTH.csv'
rr_CSV_PATH = '../data/rr.csv'
aa_CSV_PATH = '../data/aa.csv'

# 월별 물동량의 장기 성장

In [159]:
df1 = pd.read_csv(
    YEAR_CSV_PATH, encoding='utf-8')
df1

,년도,합계,자성대부두,신선대부두,감만부두,우암부두,감천한진,신감만부두,일반부두,신항1부두,신항2부두,신항3부두,신항4부두,신항5부두,다목적부두등
0,1994,48140,0,0,0,0,0,0,48140,0,0,0,0,0,0
1,1995,59775,0,622,0,0,0,0,59153,0,0,0,0,0,0
2,1996,4809217,1673341,1302513,0,21171,0,0,1812192,0,0,0,0,0,0
3,1997,5246646,1757757,1413847,23,316902,11848,0,1746269,0,0,0,0,0,0
4,1998,5701360,1096650,1121949,891424,281571,348910,0,1960856,0,0,0,0,0,0
5,1999,6555803,995119,1160577,1458472,355761,449766,0,2136108,0,0,0,0,0,0
6,2000,7729508,1304180,1281914,1887418,342486,403174,0,2510336,0,0,0,0,0,0
7,2001,8073125,1272333,1319812,1922622,447704,432972,0,2677682,0,0,0,0,0,0
8,2002,9453858,1534639,1528424,2261533,502460,504269,481202,2641331,0,0,0,0,0,0
9,2003,10408238,1584468,1786165,2546451,533298,512285,745613,2699958,0,0,0,0,0,0


In [160]:
df1.describe()

,년도,합계,자성대부두,신선대부두,감만부두,우암부두,감천한진,신감만부두,일반부두,신항1부두,신항2부두,신항3부두,신항4부두,신항5부두,다목적부두등
count,31.000000,3.100000e+01,3.100000e+01,3.100000e+01,3.100000e+01,31.000000,31.000000,3.100000e+01,3.100000e+01,3.100000e+01,3.100000e+01,3.100000e+01,3.100000e+01,3.100000e+01,3.100000e+01
mean,2009.000000,1.396878e+07,1.593494e+06,1.913994e+06,1.560014e+06,274748.870968,174349.483871,7.506600e+05,1.291686e+06,1.123328e+06,2.123786e+06,1.238136e+06,1.027238e+06,7.676622e+05,1.296865e+05
std,9.092121,6.999841e+06,5.343120e+05,7.026255e+05,8.186354e+05,260821.050294,237488.157209,4.856746e+05,1.102136e+06,1.169519e+06,2.238415e+06,1.285071e+06,1.118048e+06,1.002712e+06,3.591327e+05
min,1994.000000,4.814000e+04,0.000000e+00,0.000000e+00,0.000000e+00,0.000000,0.000000,0.000000e+00,4.814000e+04,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00
25%,2001.500000,8.763492e+06,1.404957e+06,1.471136e+06,1.229098e+06,433.500000,623.000000,2.341395e+05,3.041670e+05,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00
50%,2009.000000,1.345451e+07,1.729414e+06,2.054646e+06,1.498394e+06,316902.000000,1880.000000,9.763480e+05,6.781770e+05,7.011240e+05,7.968300e+05,9.224190e+05,0.000000e+00,0.000000e+00,2.140000e+02
75%,2016.500000,1.998110e+07,1.920802e+06,2.388700e+06,1.999389e+06,540690.000000,441369.000000,1.126288e+06,2.575834e+06,2.419781e+06,4.578868e+06,2.511854e+06,2.202690e+06,1.740969e+06,5.044850e+04
max,2024.000000,2.440202e+07,2.275402e+06,2.813335e+06,2.862251e+06,640339.000000,574824.000000,1.250360e+06,3.202849e+06,3.018315e+06,5.501091e+06,3.076160e+06,2.560274e+06,2.496909e+06,1.844868e+06


In [161]:
df1[['년도', '합계']]

,년도,합계
0,1994,48140
1,1995,59775
2,1996,4809217
3,1997,5246646
4,1998,5701360
5,1999,6555803
6,2000,7729508
7,2001,8073125
8,2002,9453858
9,2003,10408238


In [162]:
yearsum = df1[['년도', '합계']]

In [163]:
yearsum = df1[df1['년도'].ge(1995)]

In [164]:
yearsum['합계'].pct_change().dropna() * 100

2     7945.532413
3        9.095639
4        8.666756
5       14.986652
6       17.903299
7        4.445522
8       17.102832
9       10.095138
10      10.416537
11       3.052013
12       1.650243
13      10.188222
14       1.427646
15     -11.067080
16      18.325450
17      13.549280
18       6.031338
19       3.754031
20       5.638258
21       4.203955
22      -0.063861
23       5.330856
24       5.704712
25       1.520724
26      -0.763932
27       4.042037
28      -2.765487
29       4.870484
30       5.392318
Name: 합계, dtype: float64

In [165]:
yearsum = df1[['년도', '합계']].copy()
yearsum['증감률(%)'] = yearsum['합계'].pct_change() * 100

yearsum = yearsum[yearsum['년도'] >= 1997]
yearsum = yearsum[['년도', '합계', '증감률(%)']]

yearsum

,년도,합계,증감률(%)
3,1997,5246646,9.095639
4,1998,5701360,8.666756
5,1999,6555803,14.986652
6,2000,7729508,17.903299
7,2001,8073125,4.445522
8,2002,9453858,17.102832
9,2003,10408238,10.095138
10,2004,11492416,10.416537
11,2005,11843166,3.052013
12,2006,12038607,1.650243


- 장기적으로 물동량은 증가하는 추세이나, 연도별 변동이 존재함.
- 특히 2009년에는 약 11.1% 감소한 후 2010년 약 18.3% 증가하며 빠른 회복세를 보임.

# 계절성

In [166]:
df2 = pd.read_csv(
    MONTH_CSV_PATH, encoding='cp949')
df2

,연도,월,연월,수출입구분,수출입구분설명,물동량
0,2003,2,2003-02,OT,수출환적,161074.50
1,2003,11,2003-11,IT,수입환적,170733.50
2,2006,1,2006-01,OT,수출환적,203348.25
3,2018,1,2018-01,OO,수출,413797.75
4,2013,6,2013-06,OO,수출,389650.25
...,...,...,...,...,...,...
1371,1996,1,1996-01,OO,수출,158518.00
1372,2001,6,2001-06,OO,수출,215325.25
1373,2001,9,2001-09,OT,수출환적,127883.00
1374,2004,1,2004-01,IT,수입환적,181137.50


In [189]:
df2.shape

(1376, 6)

In [190]:
df2.dtypes

연도           int64
월            int64
연월             str
수출입구분          str
수출입구분설명        str
물동량        float64
dtype: object

In [191]:
df2.info()

<class 'pandas.DataFrame'>
RangeIndex: 1376 entries, 0 to 1375
Data columns (total 6 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   연도       1376 non-null   int64  
 1   월        1376 non-null   int64  
 2   연월       1376 non-null   str    
 3   수출입구분    1376 non-null   str    
 4   수출입구분설명  1376 non-null   str    
 5   물동량      1376 non-null   float64
dtypes: float64(1), int64(2), str(3)
memory usage: 64.6 KB


In [192]:
df2.describe()

,연도,월,물동량
count,1376.000000,1376.000000,1376.000000
mean,2009.982558,6.470930,324784.228924
std,8.861585,3.464399,119286.831230
min,1992.000000,1.000000,90300.000000
25%,2003.000000,3.000000,222387.250000
50%,2010.000000,6.000000,322195.500000
75%,2017.250000,9.000000,420758.437500
max,2025.000000,12.000000,635437.000000


In [193]:
df2.isna().sum()

연도         0
월          0
연월         0
수출입구분      0
수출입구분설명    0
물동량        0
dtype: int64

In [167]:
df2_sorted = df2.sort_values(by='연월', ascending=True).reset_index(drop=True)

df2_sorted

,연도,월,연월,수출입구분,수출입구분설명,물동량
0,1992,1,1992-01,OO,수출,104567.00
1,1992,1,1992-01,II,수입,92005.00
2,1992,2,1992-02,OO,수출,103903.00
3,1992,2,1992-02,II,수입,91669.00
4,1992,3,1992-03,II,수입,105276.00
...,...,...,...,...,...,...
1371,2025,1,2025-01,OO,수출,434429.50
1372,2025,2,2025-02,OT,수출환적,599626.75
1373,2025,2,2025-02,OO,수출,437906.75
1374,2025,2,2025-02,II,수입,392441.25


In [168]:
year_month = df2.groupby(['연도', '월'])['물동량'].sum().reset_index()
year_month

,연도,월,물동량
0,1992,1,196572.00
1,1992,2,195572.00
2,1992,3,242358.00
3,1992,4,229614.00
4,1992,5,222359.00
...,...,...,...
393,2024,10,2049390.75
394,2024,11,1998100.75
395,2024,12,2067420.25
396,2025,1,2126084.75


In [169]:
year_avg = year_month.groupby('연도')['물동량'].transform('mean')

In [170]:
year_month['평균대비(%)'] = (
    year_month['물동량'] / year_avg * 100
)

year_month

,연도,월,물동량,평균대비(%)
0,1992,1,196572.00,89.507751
1,1992,2,195572.00,89.052408
2,1992,3,242358.00,110.356101
3,1992,4,229614.00,104.553206
4,1992,5,222359.00,101.249690
...,...,...,...,...
393,2024,10,2049390.75,100.781366
394,2024,11,1998100.75,98.259115
395,2024,12,2067420.25,101.667989
396,2025,1,2126084.75,103.974475


In [171]:
monthly_pattern = year_month.groupby('월')['평균대비(%)'].mean()

monthly_pattern

월
1      94.749937
2      88.573391
3     103.738077
4     101.925334
5     103.329926
6     100.226616
7     102.654918
8      99.384748
9      98.094913
10    102.509389
11    101.489932
12    103.828171
Name: 평균대비(%), dtype: float64

In [172]:
monthly = (
df2.groupby(['연도', '월'])['물동량'].sum().reset_index()
)

monthly_avg = (monthly.groupby('월')['물동량'].mean().reset_index()
)

monthly_avg['증감률(%)'] = monthly_avg['물동량'].pct_change() * 100

monthly_avg

,월,물동량,증감률(%)
0,1,1.104992e+06,NaN
1,2,1.029117e+06,-6.866582
2,3,1.161059e+06,12.820883
3,4,1.143037e+06,-1.552130
4,5,1.156499e+06,1.177677
5,6,1.122079e+06,-2.976181
6,7,1.148085e+06,2.317646
7,8,1.111876e+06,-3.153868
8,9,1.088756e+06,-2.079380
9,10,1.140525e+06,4.754930


In [173]:
avg = monthly_avg['물동량'].mean()

monthly_avg['평균차이'] = monthly_avg['물동량'] - avg

In [174]:
monthly_avg['차이율(%)'] = (monthly_avg['평균차이'] / avg * 100).round(2)
monthly_avg

,월,물동량,증감률(%),평균차이,차이율(%)
0,1,1.104992e+06,NaN,-18162.111668,-1.62
1,2,1.029117e+06,-6.866582,-94037.288139,-8.37
2,3,1.161059e+06,12.820883,37904.567253,3.37
3,4,1.143037e+06,-1.552130,19883.430890,1.77
4,5,1.156499e+06,1.177677,33344.718769,2.97
5,6,1.122079e+06,-2.976181,-1074.781231,-0.10
6,7,1.148085e+06,2.317646,24931.044526,2.22
7,8,1.111876e+06,-3.153868,-11278.046383,-1.00
8,9,1.088756e+06,-2.079380,-34398.167595,-3.06
9,10,1.140525e+06,4.754930,17371.408163,1.55


- 월별 물동량은 전체 평균을 기준으로 일정한 차이를 보이고 있음
- 특히 2월은 전체 월평균보다 -8.37%로 가장 낮고, 9월 또한 -3.06%로 낮음
- 반면에 3월은 3.37%로 가장 높게 나타남

# 환적 구조 파악

In [184]:
df3 = pd.read_csv(
    rr_CSV_PATH, encoding='cp949')
df3

,연도,컨물동량,증감율,수출입소계,수입,수출,환적소계,환적점유율,환적증감율
0,2013,17686,3.75,8934,4424,4509,8748,49.47,7.38
1,2014,18683,5.64,9254,4596,4658,9429,50.47,7.78
2,2015,19469,4.20,9364,4714,4650,10105,51.91,7.17
3,2016,19456,-0.06,9620,4801,4819,9836,50.55,-2.67
4,2017,20493,5.33,10186,5042,5144,10225,49.90,3.96
5,2018,21663,5.70,10233,5117,5116,11429,52.76,11.77
6,2019,21992,1.52,10354,5192,5162,11638,52.92,1.83
7,2020,21824,-0.76,9804,4853,4951,12020,55.08,3.28
8,2021,22706,4.04,10433,5208,5225,12273,54.05,2.10
9,2022,22078,-2.77,10311,5133,5178,11766,53.29,-4.13


- 부산항의 전체 물동량이 증가하는 가운데, 수출입물동량보다 환적물동량이 빠르게 증가한다는 것을 알 수 있음.

In [185]:
print(df3.columns.tolist())

['연도', ' 컨물동량', '증감율', '수출입소계', '수입', '수출', '환적소계', '환적점유율', '환적증감율']


In [187]:
result = df3[[
    '연도',
    ' 컨물동량',
    '수출입소계',
    '환적소계',
    '환적점유율'
]]

result

,연도,컨물동량,수출입소계,환적소계,환적점유율
0,2013,17686,8934,8748,49.47
1,2014,18683,9254,9429,50.47
2,2015,19469,9364,10105,51.91
3,2016,19456,9620,9836,50.55
4,2017,20493,10186,10225,49.90
5,2018,21663,10233,11429,52.76
6,2019,21992,10354,11638,52.92
7,2020,21824,9804,12020,55.08
8,2021,22706,10433,12273,54.05
9,2022,22078,10311,11766,53.29


In [204]:
df3['전체증가량'] = df3[' 컨물동량'].diff()
df3['환적증가량'] = df3['환적소계'].diff()

df3['환적기여율'] = (df3['환적증가량'] / df3['전체증가량'] * 100).round(2)

df3[['연도', ' 컨물동량', '전체증가량', '환적증가량', '환적점유율', '환적기여율']]

,연도,컨물동량,전체증가량,환적증가량,환적점유율,환적기여율
0,2013,17686,NaN,NaN,49.47,NaN
1,2014,18683,997.0,681.0,50.47,68.30
2,2015,19469,786.0,676.0,51.91,86.01
3,2016,19456,-13.0,-269.0,50.55,2069.23
4,2017,20493,1037.0,389.0,49.90,37.51
5,2018,21663,1170.0,1204.0,52.76,102.91
6,2019,21992,329.0,209.0,52.92,63.53
7,2020,21824,-168.0,382.0,55.08,-227.38
8,2021,22706,882.0,253.0,54.05,28.68
9,2022,22078,-628.0,-507.0,53.29,80.73
